src/streamlink/plugins/youtube.py

```diff
@@ -288,6 +334,8 @@ def _get_stream_info(self, url):
                 query_info = dict(parse_qsl(urlparse(url).query))
                 if "channel" in query_info:
                     video_id = self._get_channel_video(query_info["channel"])
+        
+        self.video_id = video_id
```

Comment:
**This was only meant for testing,**

If you call `set_title_info` without `_get_streams()`, you will get a _NameError_

This needs an update when the **main feature is done**.

Reply:
Hmmm, I'll try to make that not happen. I didn't think anyone would want to do that.